### ISSUE TITLE
Support for bit-level operations

### ISSUE THREAD

**Issue by ia0** [xldenis/creusot#667]
Hi,

I wanted to try proving the correctness of [`data-encoding`](https://crates.io/crates/data-encoding) but I hit the following error:

```
error[creusot]: unsupported binary operation: Shl
```

Is support for bit-level operations planned? I could not find an existing issue about it. Would be happy to subscribe to such issue and give it another try once there's support.

Thanks!

**Comment by xldenis** [comment]
> Is support for bit-level operations planned? 

Yes, but with no firm date yet. It actually is already possible to some degree by swapping out the relevant modules in the `prelude`. 

> Would be happy to subscribe to such issue and give it another try once there's support.

Let's consider this the issue then!

**Comment by jhjourdan** [comment]
This is a duplicate of #743.

**Comment by ia0** [comment]
It's more of the opposite :-) but I'll subscribe to the other one then. Thanks for the pointer and looking into it!

**Comment by jhjourdan** [comment]
Well, the other one has a longer discussion.
